var/spack/repos/builtin/packages/hip/package.py

```diff
@@ -69,24 +69,34 @@ def get_rocm_prefix_info(self):
                 msg += "manually edit fallback_prefix in the package file as "
                 msg += "a workaround."
                 raise RuntimeError(msg)
-
-            return {
-                'rocm-path': fallback_prefix,
-                'llvm-amdgpu': fallback_prefix.llvm,
-                'hsa-rocr-dev': fallback_prefix.hsa,
-                'rocminfo': fallback_prefix.bin,
-                'rocm-device-libs': fallback_prefix.lib,
-                'device_lib_path': fallback_prefix.lib
-            }
+            # ROCm 3.9 changes the default location of the device_lib_path
+            if '@3.9.0:' in self.spec:
+                return {
+                    'rocm-path': fallback_prefix,
+                    'llvm-amdgpu': fallback_prefix.llvm,
+                    'hsa-rocr-dev': fallback_prefix.hsa,
+                    'rocminfo': fallback_prefix.bin,
+                    'rocm-device-libs': fallback_prefix.amdgcn.bitcode,
+                    'device_lib_path': fallback_prefix.amdgcn.bitcode
```

Comment:
Why do the packages depending on this use `$DEVICE_LIB_PATH` instead of the value returned here?

Reply:
> @cmakrides-hpe @dtaller I'm a little concerned that we're proliferating the same information via two mechanisms.

I think there was some issue with the hand-off between hip and its dependent package. I think previously you suggested doing the handoff via the setup_dependent_package function in hip.py to set something spec.device_path or something similar, but I am ok either way. I am not sure if Kostas tried that and ran into some issue, or if he just preferred this method.

Reply:
Within the hip package, I believe the dictionary is exclusively used. I didn't have much success with forwarding this to a dependent package and the workaround was to use the environment. At the time, I realized that the hip package exposed the dictionary via the environment and assumed the intention was for dependent packages to get the rocm paths from the environment.

Reply:
Hi @cmakrides-hpe . Could you please show us what errors arise exactly when you do something like what Greg suggested (like adding spec.extra_attributes[‘device_lib_path’] = path etc and accessing it from umpire, care, etc)?

If Greg sees exactly what errors arise in his approach he might be better equipped to offer a fix, a different suggestion, or approve the current approach you are using so we can move forward

Reply:
I'll have to re-create those attempts since I don't think I pushed a commit locally with those attempts.
It seems that this approach is the preferred method of passing these arguments?

Reply:
I have contributed to spack a bit, but I would defer to Greg and other developers who work on this full time, see what their design philosophy is exactly. But that is my impression from previous comments

Reply:
Through some trial and error I was able to use the setup_dependent_package interface and pass a path defined from the hip package. @becker33 do you have as to if I should keep the environment variables that I previously used to pass this around?

Reply:
I would recommend using the setup_dependent_package interface exclusively.

In general, I recommend using environment variables only when those are the expected interface to the package, and use the spack spec/package as the interface to convey information to Spack itself.